engine/occurrence.ftl

```diff
@@ -203,13 +203,42 @@
     [#return occurrence]
 [/#function]
 
+[#function extendOccurrenceAttributes provider attributes=[] extensions=[]]
+    [#local result = []]
+    [#if extensions?has_content]
+        [#local providerExtensions = extensions[provider]![]]
+        [#list attributes as attribute]
+
+            [#local attributeExtension = providerExtensions?filter(e -> asArray(attribute.Names)?seq_contains(e.Names))]
+            [#if attributeExtension?has_content]
+  
+                [#local extendedValues = combineEntities(
```

Comment:
I think we had discussed prefixing the attribute when it was being added. 

So for your example, the required value would be `azure:examplevalue` instead of `examplevalue`

I think this provides better understanding as to where the value came from, especially if I was to copy a component definition from one solution to another where the new solution was aws based instead of azure and it started throwing an error that the value provided was incorrect

Reply:
You're right, we had talked about that! I have added it in cb51fca and testing shows it working as intended

```json
{
  "Timestamp": "2020-09-10T00:44:22.984Z",
  "Severity": "fatal",
  "Message": "Attribute value is not one of the expected values",
  "Context": {
    "Name": "Role",
    "Value": "mockprovider:examplevalue",
    "ExpectedValues": [
      "appdata",
      "operations",
      "staticWebsite",
      "azure:examplevalue"
    ]
 ...
}
```